File: BehaviorPattern/State/Java/Contexts/PhoneContext.java

package Contexts;

import Interfaces.PhoneState;

public class PhoneContext {
    private PhoneState currentState;

    public PhoneContext(PhoneState initialState) {
        this.currentState = initialState;
    }

    public void setState(PhoneState state) {
        this.currentState = state;
    }

    public void pressButton() {
        currentState.pressButton(this);
    }
}

File: BehaviorPattern/State/Python/Contexts/PhoneContext.py

from Interfaces import PhoneState

class PhoneContext:
    def __init__(self, initial_state: PhoneState):
        self.current_state = initial_state

    def set_state(self, state: PhoneState):
        self.current_state = state

    def press_button(self):
        self.current_state.press_button(self)
